docs/timezone.md

```diff
@@ -4,8 +4,6 @@ A `Temporal.TimeZone` is a representation of a time zone: either an [IANA time z
 
 Since `Temporal.Absolute` and `Temporal.DateTime` do not contain any time zone information, a `Temporal.TimeZone` object is required to convert between the two.
 
-Finally, the `Temporal.TimeZone` object itself provides access to a list of the time zones in the IANA time zone database.
```

Comment:
Why is this being removed? This seems like very very important data to expose. What if I want to populate a dropdown?

Reply:
@ptomato explains in tc39/ecma402#435

tl;dr is that this API is a fingerprinting vector, so we shouldn't add it without good reason.  The dropdown menu is an acknowledged use case, but that use case would also require time zone display names in Intl, not just a list of IANA names (e.g., localized and human-readable strings like "US Central" instead of machine-readable "America/Chicago").  Adding time zone display names to Intl is another can of worms that we don't want to open right now.  Also, if we expose a list of time zones, then maybe we should expose a list of calendars, locales, et cetera, and think about those things as one.  Therefore, it's better to think about this as a separate task that can be done in a follow-up proposal.

Reply:
for one, human-readable names are not necessarily required; as for fingerprinting vectors, that ship seems like it's totally sailed - and regardless, i can already do the *exact same fingerprinting* with a manual list of timezones that i iterate over and see which the Temporal impl supports.

Reply:
The best practice in exposing a list of time zones is to include localized display names.  Maybe not every application will use them, but by giving a list of IANA names without a way to translate them, we encourage a worse practice in general.

It's not the *exact* same fingerprint, because the implementation might support time zones that aren't in your test list.

However, the most compelling reason for me to hold back on this feature is that we should think about how this relates to the lists of calendars, languages, regions, currencies, and other things that JavaScript natively supports.  Time zones aren't any more special than those other things.  We should figure out a way to expose implementation-dependent lists of identifiers in a cohesive way.

Reply:
> The best practice in exposing a list of time zones is to include localized display names. Maybe not every application will use them, but by giving a list of IANA names without a way to translate them, we encourage a worse practice in general.

This is also reinforced by the tzdb documentation. There have been many, many discussions on the mailing list of problems caused by exposing the internal ids to end users.

As described in the [tzdb theory page](https://data.iana.org/time-zones/theory.html):

> Each timezone has a name that uniquely identifies the timezone. Inexperienced users are not expected to select these names unaided. Distributors should provide documentation and/or a simple selection interface that explains each name via a map or via descriptive text like "Ruthenia" instead of the timezone name "`Europe/Uzhgorod`".

Reply:
FYI, this discussion has moved to https://github.com/tc39/ecma402/issues/435